pkg/mig/config/config_test.go

```diff
@@ -23,31 +23,33 @@ import (
 
 	"github.com/stretchr/testify/require"
 
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
+	"github.com/NVIDIA/go-nvml/pkg/nvml/mock/dgxa100"
+
 	"github.com/NVIDIA/mig-parted/internal/nvlib"
-	"github.com/NVIDIA/mig-parted/internal/nvml"
 	"github.com/NVIDIA/mig-parted/pkg/types"
 )
 
 func NewMockLunaServerMigConfigManager() Manager {
-	nvml := nvml.NewMockNVMLOnLunaServer()
+	nvml := dgxa100.New()
 	nvlib := nvlib.NewMock(nvml)
 	return &nvmlMigConfigManager{nvml, nvlib}
 }
 
 func EnableMigMode(manager Manager, gpu int) (nvml.Return, nvml.Return) {
 	m := manager.(*nvmlMigConfigManager)
-	n := m.nvml.(*nvml.MockLunaServer)
+	n := m.nvml.(*dgxa100.Server)
 	r1, r2 := n.Devices[gpu].SetMigMode(nvml.DEVICE_MIG_ENABLE)
 	return r1, r2
 }
 
 func TestGetSetMigConfig(t *testing.T) {
-	nvmlLib := nvml.NewMockNVMLOnLunaServer()
+	nvmlLib := dgxa100.New()
 	manager := NewMockLunaServerMigConfigManager()
```

Comment:
Nit: We instantiate an `nvmlLib` twice here since we also call `dgxa100.New()` in `NewMockLunaServerMigConfigManager`. Does it make sense to inject it or return it from the function?

Reply:
Possibly, but this issue predates the changes here, and I don't want to make that change as part of this PR. This PR should be scoped to only pull in changes 1:1 with the updated library.